runtime/gc_vlhgc/IncrementalGenerationalGC.cpp

```diff
@@ -1351,12 +1381,14 @@ MM_IncrementalGenerationalGC::partialGarbageCollectUsingCopyForward(MM_Environme
 
 	/* It is possible that we could end up with evacuate regions which were not compacted (inaccurate RSCL) so we need to detect that case and sweep such regions, before completing the PGC */
 	UDATA regionsSkippedByCompactorRequiringSweep = 0;
-	if (useSlidingCompactor) {
+	if (static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_useSlidingCompactor) {
 		/* compact to meet compaction targets, as well as compacting any unsuccessfully evacuated regions */
+		uintptr_t desiredCompactWork = static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_desiredCompactWork;
 		_reclaimDelegate.runCompact(env, allocDescription, env->_cycleState->_activeSubSpace, desiredCompactWork, env->_cycleState->_gcCode, _markMapManager->getGlobalMarkPhaseMap(), &regionsSkippedByCompactorRequiringSweep);
 		/* we can't tell exactly how many bytes were added to meet the compact goal, so we'll assume it was an exact match. The precise amount could be lower or higher */ 
 		static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_vlhgcIncrementStats._copyForwardStats._externalCompactBytes = desiredCompactWork;
-	} else if(!successful || _copyForwardDelegate.isHybrid(env)) {
+	/* static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_abortFlagRaisedDuringPGC dictates if PGC was indeed successful or not */
+	} else if(!static_cast<MM_CycleStateVLHGC*>(env->_cycleState)->_abortFlagRaisedDuringPGC || _copyForwardDelegate.isHybrid(env)) {
```

Comment:
put _cycleState in a local (for easier reading), it's being used several times in this method (probably makes sense for preProcess, too)

there is also missing space between 'if('